### PR TITLE
`tqdm.contrib.discord`: replace `disco-py` with `requests`

### PR DESCRIPTION
**Pull request by guigoruiz1**
This PR is an alternative to PR #1383 with the preserved functionality of choosing to which `channel_id` to post the progress bar preserved. This is entirely built up on Discord REST API and requires no additional packages to run. This is analogous to the Telegram contrib.

### CODE REVIEW

**Comment by guigoruiz1** [comment]
@nzjrs my implementation is exactly the same as the implementation for the telegram contrib in the base branch. I can, obviously, improve it with your suggestion, but I didn't want to deviate much from the way the maintainers here have been doing it. I wouldn't call it mediocre though.

**Review by casperdcl — APPROVED** [review]
Sorry only just saw this... thanks @guigoruiz1! Ages ago I couldn't find any Discord API docs so didn't implement it this way. Taking another look now; it seems pretty clear:

- https://discord.com/developers/docs/reference#http-api
- https://discord.com/developers/docs/resources/message

I'll rebase & merge this soon after #1627

**Pull request merged**
